spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubTemplateIntegrationTests.java

```diff
@@ -160,11 +160,16 @@ public void testPullAndAck() {
 			assertThat(messagesSet.size()).as("check that we received all the messages").isEqualTo(3);
 
 			ackableMessages.forEach((message) -> {
-				if (message.getPubsubMessage().getData().toStringUtf8().equals("message1")) {
-					message.ack(); //sync call
+				try {
+					if (message.getPubsubMessage().getData().toStringUtf8().equals("message1")) {
+						message.ack().get(); // sync call
+					}
+					else {
+						message.nack().get(); // sync call
+					}
 				}
-				else {
-					message.nack(); //sync call
+				catch (InterruptedException | ExecutionException e) {
+					e.printStackTrace();
```

Comment:
`e.printStackTrace();` is not good.
You should log error or re-throw.
Also never swallow `InterruptedException`. At least cal `Thread.currentThread().interrupt()` in this `catch` block.

Reply:
Done.